pysmt/test/test_hr_parsing.py

```diff
@@ -0,0 +1,90 @@
+#
+# This file is part of pySMT.
+#
+#   Copyright 2014 Andrea Micheli and Marco Gario
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+import unittest
+import tempfile, os
+
+from pysmt.parsing import HRParser
+from pysmt.shortcuts import is_valid, Iff, Symbol, And, Or, LE, Real, Plus, Minus
+from pysmt.test.examples import get_example_formulae
+from pysmt.exceptions import NoSolverAvailableError
+from pysmt.test import TestCase
+from pysmt.typing import REAL
+
+class TestHRParser(TestCase):
+
+    def test_precedences(self):
+        p = HRParser()
+
+        a,b,c = (Symbol(v) for v in "abc")
+        x,y = (Symbol(v, REAL) for v in "xy")
+
+        tests = []
```

Comment:
This is not better than writing multiple assertions. When/If a test fails, you will not get the exact line that makes the test fail, but you will always get line 49. Printing additional info in the assertEqual mitigates the issue, but I do not see the need for this list in the first place.